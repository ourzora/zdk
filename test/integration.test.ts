import { ZDK } from '../src/index';
import { CollectionSortKey, SortDirection } from '../src/queries/queries-sdk';

describe('zdk', () => {
  let zdk: ZDK;
  beforeEach(() => {
    zdk = new ZDK();
  });
  it('should fetch localhost collections empty object', async () => {
    const apiResult = await zdk.tokens({
      where: {
        collectionAddresses: ['0xCa21d4228cDCc68D4e23807E5e370C07577Dd152'],
      },
      pagination: {
        limit: 2,
      },
    });
    expect(apiResult.tokens.nodes[0]).toMatchSnapshot();
    expect(apiResult.tokens.nodes[1]).toMatchSnapshot();
    expect(apiResult.tokens.nodes.length).toBe(2);
    // 20 sec timeout
  }, 20_000);
  it('should fetch localhost token full object', async () => {
    expect(
      await zdk.tokens({
        where: {
          tokens: [
            {
              address: '0xCa21d4228cDCc68D4e23807E5e370C07577Dd152',
              tokenId: '48057',
            },
          ],
        },
        includeFullDetails: true,
        includeSalesHistory: false,
      })
    ).toMatchSnapshot();
  });
  it('queries sort collection information', async () => {
    expect(
      await zdk.collections({
        where: {
          collectionAddresses: ['0xCa21d4228cDCc68D4e23807E5e370C07577Dd152'],
        },
        sort: { sortKey: CollectionSortKey.None, sortDirection: SortDirection.Asc },
      })
    ).toMatchSnapshot();
  });
  it('queries long token information', async () => {
    expect(
      await zdk.collections({
        where: {
          collectionAddresses: ['0x5180db8f5c931aae63c74266b211f580155ecac8'],
        },
        sort: { sortKey: CollectionSortKey.None, sortDirection: SortDirection.Asc },
      })
    ).toMatchSnapshot();
  });
});
