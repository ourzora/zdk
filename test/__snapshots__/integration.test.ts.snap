// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zdk queries long token information 1`] = `
Object {
  "collections": Object {
    "__typename": "CollectionConnection",
    "nodes": Array [
      Object {
        "address": "0x5180db8f5c931aae63c74266b211f580155ecac8",
        "description": "",
        "name": "Crypto Coven",
        "networkInfo": Object {
          "chain": "MAINNET",
          "network": "ETHEREUM",
        },
        "symbol": "WITCH",
        "totalSupply": null,
      },
    ],
    "pageInfo": Object {
      "endCursor": "eyJza2lwIjogNTAwfQ==",
      "hasNextPage": false,
      "limit": 500,
    },
  },
}
`;

exports[`zdk queries sort collection information 1`] = `
Object {
  "collections": Object {
    "__typename": "CollectionConnection",
    "nodes": Array [
      Object {
        "address": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
        "description": "",
        "name": "Zorbs",
        "networkInfo": Object {
          "chain": "MAINNET",
          "network": "ETHEREUM",
        },
        "symbol": "ZORB",
        "totalSupply": 56741,
      },
    ],
    "pageInfo": Object {
      "endCursor": "eyJza2lwIjogNTAwfQ==",
      "hasNextPage": false,
      "limit": 500,
    },
  },
}
`;

exports[`zdk should fetch localhost collections empty object 1`] = `
Object {
  "__typename": "TokenWithMarketsSummary",
  "marketsSummary": Array [],
  "token": Object {
    "__typename": "Token",
    "collectionAddress": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
    "content": Object {
      "mediaEncoding": Object {
        "__typename": "UnsupportedEncodingTypes",
        "original": "https://api.zora.co/media/ETHEREUM-MAINNET/0xca21d4228cdcc68d4e23807e5e370c07577dd152/1/content/original",
      },
      "mimeType": null,
      "size": null,
      "url": null,
    },
    "description": "Zorbs were distributed for free by ZORA on New Year’s 2022. Each NFT imbues the properties of its wallet holder, and when sent to someone else, will transform.

View this NFT at [zorb.dev/nft/1](https://zorb.dev/nft/1)",
    "image": Object {
      "mediaEncoding": Object {
        "__typename": "ImageEncodingTypes",
        "large": "https://api.zora.co/media/ETHEREUM-MAINNET/0xca21d4228cdcc68d4e23807e5e370c07577dd152/1/image/large",
        "original": "https://api.zora.co/media/ETHEREUM-MAINNET/0xca21d4228cdcc68d4e23807e5e370c07577dd152/1/image/original",
        "poster": "https://api.zora.co/media/ETHEREUM-MAINNET/0xca21d4228cdcc68d4e23807e5e370c07577dd152/1/image/poster",
        "thumbnail": "https://api.zora.co/media/ETHEREUM-MAINNET/0xca21d4228cdcc68d4e23807e5e370c07577dd152/1/image/thumbnail",
      },
      "mimeType": "image/svg+xml",
      "size": "860",
      "url": "data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAxMTAgMTEwIj48ZGVmcz48cmFkaWFsR3JhZGllbnQgaWQ9Imd6ciIgZ3JhZGllbnRUcmFuc2Zvcm09InRyYW5zbGF0ZSg2Ni40NTc4IDI0LjM1NzUpIHNjYWxlKDc1LjI5MDgpIiBncmFkaWVudFVuaXRzPSJ1c2VyU3BhY2VPblVzZSIgcj0iMSIgY3g9IjAiIGN5PSIwJSI+PHN0b3Agb2Zmc2V0PSIxNS42MiUiIHN0b3AtY29sb3I9ImhzbCgxMjIsIDcyJSwgOTAlKSIgLz48c3RvcCBvZmZzZXQ9IjM5LjU4JSIgc3RvcC1jb2xvcj0iaHNsKDEyMSwgODAlLCA4MCUpIiAvPjxzdG9wIG9mZnNldD0iNzIuOTIlIiBzdG9wLWNvbG9yPSJoc2woMTE1LCA5MiUsIDYyJSkiIC8+PHN0b3Agb2Zmc2V0PSI5MC42MyUiIHN0b3AtY29sb3I9ImhzbCgxMTMsIDk3JSwgNDYlKSIgLz48c3RvcCBvZmZzZXQ9IjEwMCUiIHN0b3AtY29sb3I9ImhzbCgxMTIsIDk3JSwgNDUlKSIgLz48L3JhZGlhbEdyYWRpZW50PjwvZGVmcz48ZyB0cmFuc2Zvcm09InRyYW5zbGF0ZSg1LDUpIj48cGF0aCBkPSJNMTAwIDUwQzEwMCAyMi4zODU4IDc3LjYxNDIgMCA1MCAwQzIyLjM4NTggMCAwIDIyLjM4NTggMCA1MEMwIDc3LjYxNDIgMjIuMzg1OCAxMDAgNTAgMTAwQzc3LjYxNDIgMTAwIDEwMCA3Ny42MTQyIDEwMCA1MFoiIGZpbGw9InVybCgjZ3pyKSIgLz48cGF0aCBzdHJva2U9InJnYmEoMCwwLDAsMC4wNzUpIiBmaWxsPSJ0cmFuc3BhcmVudCIgc3Ryb2tlLXdpZHRoPSIxIiBkPSJNNTAsMC41YzI3LjMsMCw0OS41LDIyLjIsNDkuNSw0OS41Uzc3LjMsOTkuNSw1MCw5OS41UzAuNSw3Ny4zLDAuNSw1MFMyMi43LDAuNSw1MCwwLjV6IiAvPjwvZz48L3N2Zz4=",
    },
    "lastRefreshTime": "2022-05-18T04:37:19.404000",
    "mintInfo": Object {
      "mintContext": Object {
        "__typename": "TransactionInfo",
        "blockNumber": 13917526,
        "blockTimestamp": "2022-01-01T05:00:10",
        "logIndex": 57,
        "transactionHash": "0xf46f3540ebfca6fd50eac66c8e600f33e53ad891cd47032e23ef488e62b5f72e",
      },
      "originatorAddress": "0x436a257ab8f9d8579594f493ffb81fa5245ada26",
      "price": Object {
        "__typename": "PriceAtTime",
        "blockNumber": 13917526,
        "chainTokenPrice": Object {
          "decimal": 0,
          "raw": "0",
        },
        "nativePrice": Object {
          "currency": Object {
            "address": "0x0000000000000000000000000000000000000000",
            "decimals": 18,
            "name": "ETH",
          },
          "decimal": 0,
          "raw": "0",
        },
        "usdcPrice": Object {
          "decimal": 0,
          "raw": "0",
        },
      },
      "toAddress": "0x436a257ab8f9d8579594f493ffb81fa5245ada26",
    },
    "name": "Zorb #1",
    "owner": "0x436a257ab8f9d8579594f493ffb81fa5245ada26",
    "tokenContract": Object {
      "chain": 1,
      "collectionAddress": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
      "description": null,
      "name": "Zorbs",
      "network": "ETHEREUM",
      "symbol": "ZORB",
    },
    "tokenId": "1",
  },
}
`;

exports[`zdk should fetch localhost collections empty object 2`] = `
Object {
  "__typename": "TokenWithMarketsSummary",
  "marketsSummary": Array [],
  "token": Object {
    "__typename": "Token",
    "collectionAddress": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
    "content": Object {
      "mediaEncoding": Object {
        "__typename": "UnsupportedEncodingTypes",
        "original": "https://api.zora.co/media/ETHEREUM-MAINNET/0xca21d4228cdcc68d4e23807e5e370c07577dd152/4/content/original",
      },
      "mimeType": null,
      "size": null,
      "url": null,
    },
    "description": "Zorbs were distributed for free by ZORA on New Year’s 2022. Each NFT imbues the properties of its wallet holder, and when sent to someone else, will transform.

View this NFT at [zorb.dev/nft/4](https://zorb.dev/nft/4)",
    "image": Object {
      "mediaEncoding": Object {
        "__typename": "ImageEncodingTypes",
        "large": "https://api.zora.co/media/ETHEREUM-MAINNET/0xca21d4228cdcc68d4e23807e5e370c07577dd152/4/image/large",
        "original": "https://api.zora.co/media/ETHEREUM-MAINNET/0xca21d4228cdcc68d4e23807e5e370c07577dd152/4/image/original",
        "poster": "https://api.zora.co/media/ETHEREUM-MAINNET/0xca21d4228cdcc68d4e23807e5e370c07577dd152/4/image/poster",
        "thumbnail": "https://api.zora.co/media/ETHEREUM-MAINNET/0xca21d4228cdcc68d4e23807e5e370c07577dd152/4/image/thumbnail",
      },
      "mimeType": "image/svg+xml",
      "size": "860",
      "url": "data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAxMTAgMTEwIj48ZGVmcz48cmFkaWFsR3JhZGllbnQgaWQ9Imd6ciIgZ3JhZGllbnRUcmFuc2Zvcm09InRyYW5zbGF0ZSg2Ni40NTc4IDI0LjM1NzUpIHNjYWxlKDc1LjI5MDgpIiBncmFkaWVudFVuaXRzPSJ1c2VyU3BhY2VPblVzZSIgcj0iMSIgY3g9IjAiIGN5PSIwJSI+PHN0b3Agb2Zmc2V0PSIxNS42MiUiIHN0b3AtY29sb3I9ImhzbCgyODgsIDcyJSwgODQlKSIgLz48c3RvcCBvZmZzZXQ9IjM5LjU4JSIgc3RvcC1jb2xvcj0iaHNsKDI4OCwgNzglLCA3NCUpIiAvPjxzdG9wIG9mZnNldD0iNzIuOTIlIiBzdG9wLWNvbG9yPSJoc2woMjQzLCA4NSUsIDU1JSkiIC8+PHN0b3Agb2Zmc2V0PSI5MC42MyUiIHN0b3AtY29sb3I9ImhzbCgyMzgsIDg4JSwgMzklKSIgLz48c3RvcCBvZmZzZXQ9IjEwMCUiIHN0b3AtY29sb3I9ImhzbCgyMzgsIDg4JSwgMzglKSIgLz48L3JhZGlhbEdyYWRpZW50PjwvZGVmcz48ZyB0cmFuc2Zvcm09InRyYW5zbGF0ZSg1LDUpIj48cGF0aCBkPSJNMTAwIDUwQzEwMCAyMi4zODU4IDc3LjYxNDIgMCA1MCAwQzIyLjM4NTggMCAwIDIyLjM4NTggMCA1MEMwIDc3LjYxNDIgMjIuMzg1OCAxMDAgNTAgMTAwQzc3LjYxNDIgMTAwIDEwMCA3Ny42MTQyIDEwMCA1MFoiIGZpbGw9InVybCgjZ3pyKSIgLz48cGF0aCBzdHJva2U9InJnYmEoMCwwLDAsMC4wNzUpIiBmaWxsPSJ0cmFuc3BhcmVudCIgc3Ryb2tlLXdpZHRoPSIxIiBkPSJNNTAsMC41YzI3LjMsMCw0OS41LDIyLjIsNDkuNSw0OS41Uzc3LjMsOTkuNSw1MCw5OS41UzAuNSw3Ny4zLDAuNSw1MFMyMi43LDAuNSw1MCwwLjV6IiAvPjwvZz48L3N2Zz4=",
    },
    "lastRefreshTime": "2022-05-04T13:26:03.285000",
    "mintInfo": Object {
      "mintContext": Object {
        "__typename": "TransactionInfo",
        "blockNumber": 13917533,
        "blockTimestamp": "2022-01-01T05:01:01",
        "logIndex": 626,
        "transactionHash": "0x3abfdfdc7c1f2841ae7771c0522eaf1babaea28e1446345bef1db74b0c4d83c6",
      },
      "originatorAddress": "0x72c7f49ae7b664cc09f6891b68eedb08aa2fc15b",
      "price": Object {
        "__typename": "PriceAtTime",
        "blockNumber": 13917533,
        "chainTokenPrice": Object {
          "decimal": 0,
          "raw": "0",
        },
        "nativePrice": Object {
          "currency": Object {
            "address": "0x0000000000000000000000000000000000000000",
            "decimals": 18,
            "name": "ETH",
          },
          "decimal": 0,
          "raw": "0",
        },
        "usdcPrice": Object {
          "decimal": 0,
          "raw": "0",
        },
      },
      "toAddress": "0x72c7f49ae7b664cc09f6891b68eedb08aa2fc15b",
    },
    "name": "Zorb #4",
    "owner": "0x72c7f49ae7b664cc09f6891b68eedb08aa2fc15b",
    "tokenContract": Object {
      "chain": 1,
      "collectionAddress": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
      "description": null,
      "name": "Zorbs",
      "network": "ETHEREUM",
      "symbol": "ZORB",
    },
    "tokenId": "4",
  },
}
`;

exports[`zdk should fetch localhost token full object 1`] = `
Object {
  "tokens": Object {
    "__typename": "TokenWithMarketsSummaryConnection",
    "nodes": Array [
      Object {
        "__typename": "TokenWithMarketsSummary",
        "events": Array [],
        "marketsSummary": Array [
          Object {
            "collectionAddress": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
            "marketAddress": "0xce6cef2a9028e1c3b21647ae3b4251038109f42a",
            "marketType": "V3_ASK",
            "networkInfo": Object {
              "chain": "MAINNET",
              "network": "ETHEREUM",
            },
            "price": Object {
              "blockNumber": 14036921,
              "chainTokenPrice": Object {
                "decimal": 0.05,
                "raw": "50000000000000000",
              },
              "nativePrice": Object {
                "currency": Object {
                  "address": "0x0000000000000000000000000000000000000000",
                  "decimals": 18,
                  "name": "ETH",
                },
                "decimal": 0.05,
                "raw": "50000000000000000",
              },
              "usdcPrice": Object {
                "decimal": 156.40084904014552,
                "raw": "156400849040145514496",
              },
            },
            "properties": Object {
              "__typename": "V3Ask",
              "address": "0xce6cef2a9028e1c3b21647ae3b4251038109f42a",
              "askCurrency": "0x0000000000000000000000000000000000000000",
              "askPrice": Object {
                "blockNumber": 14036921,
                "chainTokenPrice": Object {
                  "decimal": 0.05,
                  "raw": "50000000000000000",
                },
                "nativePrice": Object {
                  "currency": Object {
                    "address": "0x0000000000000000000000000000000000000000",
                    "decimals": 18,
                    "name": "ETH",
                  },
                  "decimal": 0.05,
                  "raw": "50000000000000000",
                },
                "usdcPrice": Object {
                  "decimal": 156.40084904014552,
                  "raw": "156400849040145514496",
                },
              },
              "buyer": null,
              "collectionAddress": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
              "finder": null,
              "findersFeeBps": 100,
              "seller": "0xf90494e746be5674f3121c62a588dadd2b3d3d4e",
              "sellerFundsRecipient": "0xf90494e746be5674f3121c62a588dadd2b3d3d4e",
              "v3AskStatus": "CANCELED",
            },
            "status": "CANCELED",
            "transactionInfo": Object {
              "blockNumber": 14036921,
              "blockTimestamp": "2022-01-19T16:04:04",
              "logIndex": 438,
              "transactionHash": "0x7d6e8224e9a85549384a96618c7ffa7dd5c0d14e73e35c2702a45f5159fc717e",
            },
          },
          Object {
            "collectionAddress": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
            "marketAddress": "0xce6cef2a9028e1c3b21647ae3b4251038109f42a",
            "marketType": "V3_ASK",
            "networkInfo": Object {
              "chain": "MAINNET",
              "network": "ETHEREUM",
            },
            "price": Object {
              "blockNumber": 14040390,
              "chainTokenPrice": Object {
                "decimal": 0.05,
                "raw": "50000000000000000",
              },
              "nativePrice": Object {
                "currency": Object {
                  "address": "0x0000000000000000000000000000000000000000",
                  "decimals": 18,
                  "name": "ETH",
                },
                "decimal": 0.05,
                "raw": "50000000000000000",
              },
              "usdcPrice": Object {
                "decimal": 156.27094762581635,
                "raw": "156270947625816358912",
              },
            },
            "properties": Object {
              "__typename": "V3Ask",
              "address": "0xce6cef2a9028e1c3b21647ae3b4251038109f42a",
              "askCurrency": "0x0000000000000000000000000000000000000000",
              "askPrice": Object {
                "blockNumber": 14040390,
                "chainTokenPrice": Object {
                  "decimal": 0.05,
                  "raw": "50000000000000000",
                },
                "nativePrice": Object {
                  "currency": Object {
                    "address": "0x0000000000000000000000000000000000000000",
                    "decimals": 18,
                    "name": "ETH",
                  },
                  "decimal": 0.05,
                  "raw": "50000000000000000",
                },
                "usdcPrice": Object {
                  "decimal": 156.27094762581635,
                  "raw": "156270947625816358912",
                },
              },
              "buyer": null,
              "collectionAddress": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
              "finder": null,
              "findersFeeBps": 100,
              "seller": "0xf90494e746be5674f3121c62a588dadd2b3d3d4e",
              "sellerFundsRecipient": "0xf90494e746be5674f3121c62a588dadd2b3d3d4e",
              "v3AskStatus": "CANCELED",
            },
            "status": "CANCELED",
            "transactionInfo": Object {
              "blockNumber": 14040390,
              "blockTimestamp": "2022-01-20T04:57:09",
              "logIndex": 439,
              "transactionHash": "0xbc0158760124585c8589ba68c78011b1a13004f9cb20c0f24125122dec1de147",
            },
          },
        ],
        "token": Object {
          "__typename": "Token",
          "attributes": Array [],
          "collectionAddress": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
          "content": Object {
            "mediaEncoding": Object {
              "__typename": "UnsupportedEncodingTypes",
              "original": "https://api.zora.co/media/ETHEREUM-MAINNET/0xca21d4228cdcc68d4e23807e5e370c07577dd152/48057/content/original",
            },
            "mimeType": null,
            "size": null,
            "url": null,
          },
          "description": "Zorbs were distributed for free by ZORA on New Year’s 2022. Each NFT imbues the properties of its wallet holder, and when sent to someone else, will transform.

View this NFT at [zorb.dev/nft/48057](https://zorb.dev/nft/48057)",
          "image": Object {
            "mediaEncoding": Object {
              "__typename": "ImageEncodingTypes",
              "large": "https://api.zora.co/media/ETHEREUM-MAINNET/0xca21d4228cdcc68d4e23807e5e370c07577dd152/48057/image/large",
              "original": "https://api.zora.co/media/ETHEREUM-MAINNET/0xca21d4228cdcc68d4e23807e5e370c07577dd152/48057/image/original",
              "poster": "https://api.zora.co/media/ETHEREUM-MAINNET/0xca21d4228cdcc68d4e23807e5e370c07577dd152/48057/image/poster",
              "thumbnail": "https://api.zora.co/media/ETHEREUM-MAINNET/0xca21d4228cdcc68d4e23807e5e370c07577dd152/48057/image/thumbnail",
            },
            "mimeType": "image/svg+xml",
            "size": "860",
            "url": "data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAxMTAgMTEwIj48ZGVmcz48cmFkaWFsR3JhZGllbnQgaWQ9Imd6ciIgZ3JhZGllbnRUcmFuc2Zvcm09InRyYW5zbGF0ZSg2Ni40NTc4IDI0LjM1NzUpIHNjYWxlKDc1LjI5MDgpIiBncmFkaWVudFVuaXRzPSJ1c2VyU3BhY2VPblVzZSIgcj0iMSIgY3g9IjAiIGN5PSIwJSI+PHN0b3Agb2Zmc2V0PSIxNS42MiUiIHN0b3AtY29sb3I9ImhzbCgxNjMsIDcxJSwgODclKSIgLz48c3RvcCBvZmZzZXQ9IjM5LjU4JSIgc3RvcC1jb2xvcj0iaHNsKDE2MywgNzMlLCA3MSUpIiAvPjxzdG9wIG9mZnNldD0iNzIuOTIlIiBzdG9wLWNvbG9yPSJoc2woMTkwLCA3NyUsIDQ5JSkiIC8+PHN0b3Agb2Zmc2V0PSI5MC42MyUiIHN0b3AtY29sb3I9ImhzbCgxOTIsIDg3JSwgNDElKSIgLz48c3RvcCBvZmZzZXQ9IjEwMCUiIHN0b3AtY29sb3I9ImhzbCgxOTIsIDkxJSwgNDAlKSIgLz48L3JhZGlhbEdyYWRpZW50PjwvZGVmcz48ZyB0cmFuc2Zvcm09InRyYW5zbGF0ZSg1LDUpIj48cGF0aCBkPSJNMTAwIDUwQzEwMCAyMi4zODU4IDc3LjYxNDIgMCA1MCAwQzIyLjM4NTggMCAwIDIyLjM4NTggMCA1MEMwIDc3LjYxNDIgMjIuMzg1OCAxMDAgNTAgMTAwQzc3LjYxNDIgMTAwIDEwMCA3Ny42MTQyIDEwMCA1MFoiIGZpbGw9InVybCgjZ3pyKSIgLz48cGF0aCBzdHJva2U9InJnYmEoMCwwLDAsMC4wNzUpIiBmaWxsPSJ0cmFuc3BhcmVudCIgc3Ryb2tlLXdpZHRoPSIxIiBkPSJNNTAsMC41YzI3LjMsMCw0OS41LDIyLjIsNDkuNSw0OS41Uzc3LjMsOTkuNSw1MCw5OS41UzAuNSw3Ny4zLDAuNSw1MFMyMi43LDAuNSw1MCwwLjV6IiAvPjwvZz48L3N2Zz4=",
          },
          "lastRefreshTime": "2022-04-26T12:06:38.084000",
          "metadata": Object {
            "description": "Zorbs were distributed for free by ZORA on New Year’s 2022. Each NFT imbues the properties of its wallet holder, and when sent to someone else, will transform.

View this NFT at [zorb.dev/nft/48057](https://zorb.dev/nft/48057)",
            "image": "data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAxMTAgMTEwIj48ZGVmcz48cmFkaWFsR3JhZGllbnQgaWQ9Imd6ciIgZ3JhZGllbnRUcmFuc2Zvcm09InRyYW5zbGF0ZSg2Ni40NTc4IDI0LjM1NzUpIHNjYWxlKDc1LjI5MDgpIiBncmFkaWVudFVuaXRzPSJ1c2VyU3BhY2VPblVzZSIgcj0iMSIgY3g9IjAiIGN5PSIwJSI+PHN0b3Agb2Zmc2V0PSIxNS42MiUiIHN0b3AtY29sb3I9ImhzbCgxNjMsIDcxJSwgODclKSIgLz48c3RvcCBvZmZzZXQ9IjM5LjU4JSIgc3RvcC1jb2xvcj0iaHNsKDE2MywgNzMlLCA3MSUpIiAvPjxzdG9wIG9mZnNldD0iNzIuOTIlIiBzdG9wLWNvbG9yPSJoc2woMTkwLCA3NyUsIDQ5JSkiIC8+PHN0b3Agb2Zmc2V0PSI5MC42MyUiIHN0b3AtY29sb3I9ImhzbCgxOTIsIDg3JSwgNDElKSIgLz48c3RvcCBvZmZzZXQ9IjEwMCUiIHN0b3AtY29sb3I9ImhzbCgxOTIsIDkxJSwgNDAlKSIgLz48L3JhZGlhbEdyYWRpZW50PjwvZGVmcz48ZyB0cmFuc2Zvcm09InRyYW5zbGF0ZSg1LDUpIj48cGF0aCBkPSJNMTAwIDUwQzEwMCAyMi4zODU4IDc3LjYxNDIgMCA1MCAwQzIyLjM4NTggMCAwIDIyLjM4NTggMCA1MEMwIDc3LjYxNDIgMjIuMzg1OCAxMDAgNTAgMTAwQzc3LjYxNDIgMTAwIDEwMCA3Ny42MTQyIDEwMCA1MFoiIGZpbGw9InVybCgjZ3pyKSIgLz48cGF0aCBzdHJva2U9InJnYmEoMCwwLDAsMC4wNzUpIiBmaWxsPSJ0cmFuc3BhcmVudCIgc3Ryb2tlLXdpZHRoPSIxIiBkPSJNNTAsMC41YzI3LjMsMCw0OS41LDIyLjIsNDkuNSw0OS41Uzc3LjMsOTkuNSw1MCw5OS41UzAuNSw3Ny4zLDAuNSw1MFMyMi43LDAuNSw1MCwwLjV6IiAvPjwvZz48L3N2Zz4=",
            "name": "Zorb #48057",
          },
          "mintInfo": Object {
            "mintContext": Object {
              "__typename": "TransactionInfo",
              "blockNumber": 13927877,
              "blockTimestamp": "2022-01-02T19:17:42",
              "logIndex": 318,
              "transactionHash": "0x816a39f8017fe92e80d777569055a7549b5fb38b5d1c9f34bee5d940b510ed77",
            },
            "originatorAddress": "0xf90494e746be5674f3121c62a588dadd2b3d3d4e",
            "price": Object {
              "__typename": "PriceAtTime",
              "blockNumber": 13927877,
              "chainTokenPrice": Object {
                "decimal": 0,
                "raw": "0",
              },
              "nativePrice": Object {
                "currency": Object {
                  "address": "0x0000000000000000000000000000000000000000",
                  "decimals": 18,
                  "name": "ETH",
                },
                "decimal": 0,
                "raw": "0",
              },
              "usdcPrice": Object {
                "decimal": 0,
                "raw": "0",
              },
            },
            "toAddress": "0xf90494e746be5674f3121c62a588dadd2b3d3d4e",
          },
          "name": "Zorb #48057",
          "owner": "0xf90494e746be5674f3121c62a588dadd2b3d3d4e",
          "tokenContract": Object {
            "chain": 1,
            "collectionAddress": "0xca21d4228cdcc68d4e23807e5e370c07577dd152",
            "description": null,
            "name": "Zorbs",
            "network": "ETHEREUM",
            "symbol": "ZORB",
          },
          "tokenId": "48057",
          "tokenUrl": "data:application/json;base64,eyJuYW1lIjogIlpvcmIgIzQ4MDU3IiwgImRlc2NyaXB0aW9uIjogIlpvcmJzIHdlcmUgZGlzdHJpYnV0ZWQgZm9yIGZyZWUgYnkgWk9SQSBvbiBOZXcgWWVhcuKAmXMgMjAyMi4gRWFjaCBORlQgaW1idWVzIHRoZSBwcm9wZXJ0aWVzIG9mIGl0cyB3YWxsZXQgaG9sZGVyLCBhbmQgd2hlbiBzZW50IHRvIHNvbWVvbmUgZWxzZSwgd2lsbCB0cmFuc2Zvcm0uXG5cblZpZXcgdGhpcyBORlQgYXQgW3pvcmIuZGV2L25mdC80ODA1N10oaHR0cHM6Ly96b3JiLmRldi9uZnQvNDgwNTcpIiwgImltYWdlIjogImRhdGE6aW1hZ2Uvc3ZnK3htbDtiYXNlNjQsUEhOMlp5QjRiV3h1Y3owaWFIUjBjRG92TDNkM2R5NTNNeTV2Y21jdk1qQXdNQzl6ZG1jaUlIWnBaWGRDYjNnOUlqQWdNQ0F4TVRBZ01URXdJajQ4WkdWbWN6NDhjbUZrYVdGc1IzSmhaR2xsYm5RZ2FXUTlJbWQ2Y2lJZ1ozSmhaR2xsYm5SVWNtRnVjMlp2Y20wOUluUnlZVzV6YkdGMFpTZzJOaTQwTlRjNElESTBMak0xTnpVcElITmpZV3hsS0RjMUxqSTVNRGdwSWlCbmNtRmthV1Z1ZEZWdWFYUnpQU0oxYzJWeVUzQmhZMlZQYmxWelpTSWdjajBpTVNJZ1kzZzlJakFpSUdONVBTSXdKU0krUEhOMGIzQWdiMlptYzJWMFBTSXhOUzQyTWlVaUlITjBiM0F0WTI5c2IzSTlJbWh6YkNneE5qTXNJRGN4SlN3Z09EY2xLU0lnTHo0OGMzUnZjQ0J2Wm1aelpYUTlJak01TGpVNEpTSWdjM1J2Y0MxamIyeHZjajBpYUhOc0tERTJNeXdnTnpNbExDQTNNU1VwSWlBdlBqeHpkRzl3SUc5bVpuTmxkRDBpTnpJdU9USWxJaUJ6ZEc5d0xXTnZiRzl5UFNKb2Myd29NVGt3TENBM055VXNJRFE1SlNraUlDOCtQSE4wYjNBZ2IyWm1jMlYwUFNJNU1DNDJNeVVpSUhOMGIzQXRZMjlzYjNJOUltaHpiQ2d4T1RJc0lEZzNKU3dnTkRFbEtTSWdMejQ4YzNSdmNDQnZabVp6WlhROUlqRXdNQ1VpSUhOMGIzQXRZMjlzYjNJOUltaHpiQ2d4T1RJc0lEa3hKU3dnTkRBbEtTSWdMejQ4TDNKaFpHbGhiRWR5WVdScFpXNTBQand2WkdWbWN6NDhaeUIwY21GdWMyWnZjbTA5SW5SeVlXNXpiR0YwWlNnMUxEVXBJajQ4Y0dGMGFDQmtQU0pOTVRBd0lEVXdRekV3TUNBeU1pNHpPRFU0SURjM0xqWXhORElnTUNBMU1DQXdRekl5TGpNNE5UZ2dNQ0F3SURJeUxqTTROVGdnTUNBMU1FTXdJRGMzTGpZeE5ESWdNakl1TXpnMU9DQXhNREFnTlRBZ01UQXdRemMzTGpZeE5ESWdNVEF3SURFd01DQTNOeTQyTVRReUlERXdNQ0ExTUZvaUlHWnBiR3c5SW5WeWJDZ2paM3B5S1NJZ0x6NDhjR0YwYUNCemRISnZhMlU5SW5KblltRW9NQ3d3TERBc01DNHdOelVwSWlCbWFXeHNQU0owY21GdWMzQmhjbVZ1ZENJZ2MzUnliMnRsTFhkcFpIUm9QU0l4SWlCa1BTSk5OVEFzTUM0MVl6STNMak1zTUN3ME9TNDFMREl5TGpJc05Ea3VOU3cwT1M0MVV6YzNMak1zT1RrdU5TdzFNQ3c1T1M0MVV6QXVOU3czTnk0ekxEQXVOU3cxTUZNeU1pNDNMREF1TlN3MU1Dd3dMalY2SWlBdlBqd3ZaejQ4TDNOMlp6ND0ifQ==",
          "tokenUrlMimeType": "application/json",
        },
      },
    ],
    "pageInfo": Object {
      "endCursor": "eyJza2lwIjogNTAwfQ==",
      "hasNextPage": false,
      "limit": 500,
    },
  },
}
`;
